thumbnail-service/src/main/java/no/sikt/nva/thumbnail/Thumbnailer.java

```diff
@@ -0,0 +1,90 @@
+package no.sikt.nva.thumbnail;
+
+import java.io.Closeable;
+import java.io.File;
+import java.io.IOException;
+import java.net.URL;
+import java.util.List;
+
+public interface Thumbnailer extends Closeable {
```

Comment:
I would have preferred an interface with only one method like generate() with some options if needed, and rather take all input in the constructor. Unless there are reasons to be able to reuse the same thumbnail to generate thumbnails for multiple files. I would also have returned a byte array with the thumbnail rather than relying on output files.

Reply:
According to the old code, yes the input might be URL's (in which case it will spin up a browser and take a snapshot). There might be input with undetermined mime-types, in which case the thumbnailer goes into this spaghetti-code mess and check the binarystream for matching codec before sending it to the correct thumbnailer.